core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,131 @@
+  private _ayy: number;
+  private _curvatureRange: Range1d;
+  private _isValidArc: boolean;
+
+  public constructor(arc: Arc3d) {
+    const scaleData = arc.toScaledMatrix3d();
+    this._arc = Arc3d.createScaledXYColumns(scaleData.center, scaleData.axes, scaleData.r0, scaleData.r90, scaleData.sweep);
+    this._axx = arc.matrixRef.columnXMagnitudeSquared();
+    this._ayy = arc.matrixRef.columnYMagnitudeSquared();
+    this._curvatureRange = Range1d.createNull();
+    this._isValidArc = true;
+
+    if (Geometry.isSmallMetricDistanceSquared(this._axx) || Geometry.isSmallMetricDistanceSquared(this._ayy))
+      this._isValidArc = false; // ellipse must have positive radii
+    else if (Geometry.isSameCoordinateSquared(this._axx, this._ayy))
+      this._isValidArc = false; // ellipse must not be circular
+    else {
+      // extreme curvatures are at the semi-axis points
+      this._curvatureRange.extendX(Math.sqrt(this._axx) / this._ayy);
+      this._curvatureRange.extendX(Math.sqrt(this._ayy) / this._axx);
+    }
+  }
+
+  public get arc(): Arc3d {
+    return this._arc;
+  }
+  public get vector0LengthSquared(): number {
+    return this._axx;
+  }
+  public get vector90LengthSquared(): number {
+    return this._ayy;
+  }
+  public get curvatureRange(): Range1d {
+    return this._curvatureRange;
+  }
+  public get isValidArc(): boolean {
+    return this._isValidArc;
+  }
+
+  /**
+   * Compute the angle corresponding to the point in the ellipse's first quadrant with the given curvature.
+   * * The instance sweep is ignored.
+   * * Arc is assumed to be non-circular and have perpendicular axes of positive length.
+   * * This is a scaled inverse of [[Arc3d.fractionToCurvature]] restricted to fractions in [0,1/4].
+   * @return radian angle in [0,pi/2] or undefined if the ellipse is invalid, or does not attain the given curvature.
+  */
+  private curvatureToAngle(curvature: number): number | undefined {
+    if (!this.isValidArc)
+      return undefined;
+    if (!this.curvatureRange.containsX(curvature))
+      return undefined; // ellipse does not attain this curvature
+    const scaledNormalLengthSquared = this._axx * this._ayy / curvature * curvature;
+    const numerator = Math.cbrt(scaledNormalLengthSquared) - this._axx;
+    const denominator = this._ayy - this._axx;
+    const cosTheta = Math.sqrt(Math.abs(numerator / denominator));
+    return Math.acos(cosTheta);
+  }
+
+  /**
+   * Sample the instance arc by interpolating curvature, and return the fractions of the sample points.
+   * @param numPointsInQuadrant number of samples in each ellipse quadrant.
+   * @param structuredOutput false to return a flat array of fractions (default). If true, structured objects are returned:
+   * * Fractions are sorted and assembled by ellipse quadrant into a [[QuadrantFractions]] object.
+   * * If the arc sweep spans adjacent quadrants, `QuadrantFractions`representing those quadrants share the fraction at the quadrant boundary.
```

Comment:
This line was a bit vague for me. I think this makes it more clear:
```suggestion
   * * If the arc sweep spans adjacent quadrants, `QuadrantFractions` that represent those quadrants would share the fraction at the quadrant boundary.
```